Project21/Project21/Testcase.c

```diff
@@ -0,0 +1,83 @@
+// ------------------------------------------------------------------------------------------------
+// Assignment 3 ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.Naveenkumar s
+// ------------------------------------------------------------------
+// Testcase.c
+// Decimal to Binary conversion and Decimal to HexaDecimal conversion
+// ------------------------------------------------------------------------------------------------
+#include"conversionfile.h"
+
+/// <summary>Checks the user input whether its invalid or not </summary>
+int GetInteger () {
+   char input[100];
+   int number = 0;
+   char* endptr;
+   printf ("\nEnter an integer: ");
+   if (fgets (input, sizeof (input), stdin) != NULL) {
+      errno = 0;
+      number = strtol (input, &endptr, 10);
+      if (endptr == input || *endptr != '\n' && *endptr != '\0' || (number == LONG_MAX || number == LONG_MIN) && errno == ERANGE 
+          || number > INT_MAX || number <= INT_MIN || input[0] == ' ') {

```

Comment:
here in this condition (number<= int_min ), int_main should not be included as it can be represented in 32 bits.